### ISSUE TITLE
Restore Import/Export functionality from 1.0.5

### ISSUE THREAD

**Issue by RedPine91** [RimSort/RimSort#415]
The update to 1.0.6 removed import/export from the main window, and removed the ability to automatically save local modlists, all of which are the features I consider mandatory.

Solution:
1. Restore import/export to the main window.
2. Restore the ability to export to a local xml file. <SOLVED! It's called "Save mod list as" not "Export" now.>
3. Restore the ability to import from a local xml file. <SOLVED! It's called "Open mod list" not "Import" now.>

Alternatives:
I realize the importance of collecting userdata for sharing mod sorting rules, but the collection of that data should be entirely opt in.  There are many reasons a user wouldn't want to broadcast their modlist to a public and/or unsecured database.

One compromise would be allowing users to share sorting rules and mod conflicts instead of entire modlists.

Conflict hunting:
When hunting for mod conflicts, it is necessary to import and export modlists many times.  Having those options in the main window is a massive QoL feature.

Temporary Solution:
1.0.5 Was working just fine, as far as I can tell, so I'll just revert to the old version.

**Comment by hk21702** [comment]
I'm not sure what you mean by "collection of that data". RimSort doesn't collect data unless you opt in. User rules and community rules you make are saved locally and not uploaded anywhere.

If you're talking about the Rentry.co stuff, that is just the typical way with how mod lists are shared. You don't need to use it if you don't want to, and its only used when you explicitly try to use it.

I do agree that there should probably be a way to export and import from and to a local xml file for ModConfig.xml. But for now, you can just make backups of that yourself in the RimWorld config folder.

**Comment by Limfjorden** [comment]
This gave me an idea for a feature: https://github.com/RimSort/RimSort/issues/427

**Comment by RedPine91** [comment]
@LionelColaso 
Okay, that's helpful.  That lets me open (aka IMPORT) an existing modlist.  

Question: How do I EXPORT modlists to that folder? (The folder C:\Users\jtnpa\AppData\Local\RimSort)

@hk21702 Again, in 1.5 I was able to export/save/import XML modlists to that folder.  In 1.8 I have not found a way to do so.  The feature already existed, and was inexplicably removed.  (Unless I'm misremembering a Rympy feature - again, I can't check 1.5.)

---

This last one is a very minor issue.  I care more about the functionality existing than whether it takes 2 or 4 clicks to get the functionality.

@hk21702 Regarding moving the buttons off the screen and into shortcuts, I can understand your logic, but it seems weird to move one of the most frequently used buttons off of the main menu.  Most users are not shortcut/power users.  

If the buttons on the main menu are arranged vertically, there's plenty of space.  If you insist on the horizontal buttons (I've never seen horizontal buttons like that on a Windows UI before), there's still space for them where the version number is.  The version number fits at the top of the screen.

Alternatively, the refresh and restore buttons could be placed on top.  
Alternatively, the buttons could be arranged as a two row grid instead of a single row.
There are many ways to handle the UI.

**Comment by RedPine91** [comment]
Also, I would either use the same conventions as Rimpy (import for save, export for load), or use a completely different convention (save for save, load for load, share for share, etc.).  Having the same function names as Rimpy, that perform different functions, is going to be very confusing for anyone transitioning from Rimpy to Rimsort.

**Comment by bobgrey1997** [comment]
This whole feature is utterly broken. Any time I click on Refresh or reopen the program to add a new mod I've downloaded, it resets my entire mod list to whatever it was when I first opened the program back when I first downloaded it. Even saving my mod list and reloading it via the Save Mod List and Open Mod List, or trying out the copy and paste features do nothing. Worse, actually, opening my mod list doesn't open my mod list, it opens the original mod list, which was never saved. It seems this program has a hard-coded mod list that is saved when you first install and open it, and this mod list is non-changable. It will forever only read this mod list. You can make any changes, but it will get reset upon closing and reopening the program or using the refresh button.

**Comment by hk21702** [comment]
>  You can make any changes, but it will get reset upon closing and reopening the program or using the refresh button.

Did you click the save button after making changes to your mod list? @bobgrey1997

**Comment by bobgrey1997** [comment]
Yes, I did, a few hundred times. I've even fully uninstalled the program and reinstalled it, and even uninstalled all of the mods I don't want, and it STILL loads the old mod order with the mods I don't even have on my PC anymore.

**Comment by hk21702** [comment]
@bobgrey1997 make a new issue since this seems to be unrelated to what is being talked about in this issue. I presume you're saving it using the actual save button when making modifications. The one at the bottom next to sort.

**Comment by bobgrey1997** [comment]
Neither work. I click the save at the bottom, nothing happens, at all. No file, no actual saving of the list, nothing. That button is entirely useless.

I click the save as button to create a file, and it creates a file, but that files contains the default list from when I first installed the program, NOT the current list. If I make any further changes and decide to load that file to bring back my last saves list, it instead brings back the original list I had when I first installed the program.

This issues makes this program effectively only useful for auto-sorting, which the ancient in-game mod manager mod also has, which despite not having been updated for a couple years now still seems to work better than whatever this program is.

What do I even call a new ticket? "Program not working, at all, for anything it is meant to be"? You can't exactly manage mods if you can't maintain your current state, so the mod *manager* is mod managing. The mod *sorter* hardly even mod sorts because you can't save the current sorting for later. This is just an auto-sorter, it does not allow effective manual sorting.

**Comment by LionelColaso** [comment]
@bobgrey1997 can you open a new issue with all the information and details using the bug issue/report format

**Comment by hk21702** [comment]
@bobgrey1997 

1. You need to calm down. Yes it's frustrating when things doesn't work, but talking like this does not help us help you.
2. Just be specific when you are making the issue. Something like "mod list not saving" seems apt here.

**Comment by LionelColaso** [comment]
> Also, I would either use the same conventions as Rimpy (import for save, export for load), or use a completely different convention (save for save, load for load, share for share, etc.). Having the same function names as Rimpy, that perform different functions, is going to be very confusing for anyone transitioning from Rimpy to Rimsort.

This has also been fixed closing The tread as complete
